cpp/src/plasma/common.h

```diff
@@ -104,4 +96,16 @@ struct PlasmaStoreInfo;
 extern const PlasmaStoreInfo* plasma_config;
 }  // namespace plasma
 
+namespace std {
+template <>
+struct hash<::plasma::UniqueID> {
+  size_t operator()(const ::plasma::UniqueID& id) const { return id.hash(); }
+};
+
+template <>
+struct hash<const ::plasma::UniqueID> {
```

Comment:
Is it really necessary to have the two `hash<>` specializations (const and non-const)?

Reply:
Thanks for the comment. The const version is mainly for container declarations like below, without const this would not compile. This kind of usage is rare though:)

std::unordered_map<const UniqueID, int> s;

Reply:
Ok, I don't think such use makes much sense. See https://stackoverflow.com/questions/17638154/what-is-difference-between-const-and-non-const-key 
("map keys are semantically immutable" and also "some implementations re-use map nodes; under those implementations, attempting to use a const key simply won't work")

Reply:
@zhijunfu I think it makes a lot of sense to remove it for now, if there is a use case in the future, we can always re-add it. What do you think?

Reply:
@pcmoritz @pitrou sure, will remove const version.